operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -109,12 +112,6 @@ public class IngressControllerManager {
 
     protected static final String WORKER_NODE_LABEL = "node-role.kubernetes.io/worker";
 
-    /**
-     * Domain part prefixed to domain reported on IngressController status. The CNAME DNS records
-     * need to point to a sub-domain on the IngressController domain, so we just add this.
-     */
-    private static final String ROUTER_SUBDOMAIN = "ingresscontroller.";
```

Comment:
Will this change affect existing Kafkas somehow and a migration will be needed?

Reply:
@shawkins

Reply:
As coded yes we would need both the check the control plane is doing relaxed and the control plane logic to update cname records implemented.  We'll leave the do not merge label on this until we get both of these points straightened out.

Reply:
what does mean by "control plane is doing relaxed"? Also since the URL based on customer's load balancer, why should register cname in RHOSAK?

Reply:
> what does mean by "control plane is doing relaxed"?

The control plane currently validates that the returned route URL by the fleet sync corresponds to the OSD cluster dns: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/db4c66b6510a4935f793ddad6dc60952fddb0ae6/internal/kafka/internal/services/data_plane_kafka.go#L515

The OSD cluster dns is retrieved from cluster ingresses. I understand this won't be the case for private cluster and the check will have to be relaxed somehow. 

> Also since the URL based on customer's load balancer, why should register cname in RHOSAK?

If the route URL changes, KFM needs to update the cname record. From what I see, the existing router all had a `ingresscontroller.` prefix. If this URL is changed somehow, the CNAMEs record will need to be updated to point to the new routes.

Reply:
Thanks @machi1990 

On CNAME registration, what I mean is since the URL is backed by users' AWS Route 53 already (I am assuming as it is their load balancer) why also registering in RHOSAK's AWS route 53.

Reply:
@rareddy the scenario is a public kafka on an cluster that lacks a public dns entry in their config - we have already seen that this is the default state of private ROSA clusters.  If we are in that circumstance, creating cname entries that point from the application domain endpoints broker1.something to ingresscontroller.kas-zone.clusterdomain will not work - as nothing is creating a public dns entry for kas-zone.clusterdomain.  The necessary entry will point from kas-zone.clusterdomain to the aws loadbalancer hostname.  So here we are skipping that and pointing broker1.something directly to aws loabalancer hostname.  A way to do this that would address the migration concern would be leave the top level dns entries as is - and create additional route entries to point from ingresscontroller.kas-zone.clusterdomain to the aws loadbalancer hostname.  That will still require a change on the control plane side however as the target still won't be based upon the cluster domain.

Reply:
No the ingresscontroller is still used.  We still are routing traffic to the aws loadbalancer.  The aws loadbalancer has already been configured by the ingress controller operator to point to the ingress controller service / ha proxy pods.  

We can only get away from using the ingress controllers if we use loadbalancer services or switch to using ingress and the https://github.com/kubernetes-sigs/aws-load-balancer-controller - I didn't see that the operator for this is running by default, let me do some more testing on a local instance.